iothub_service_client/tests/iothub_srv_client_auth_ut/iothub_service_client_auth_ut.cpp

```diff
@@ -892,6 +896,93 @@ TEST_FUNCTION(IoTHubServiceClientAuth_CreateFromConnectionString_do_clean_up_if_
+TEST_FUNCTION(IoTHubServiceClientAuth_CreateFromConnectionString_for_sharedAccessSignature)
+{
+    // arrange
+    CIoTHubServiceClientAuthMocks mocks;
+
+    whenShallmalloc_fail = 0;
+    STRICT_EXPECTED_CALL(mocks, gballoc_malloc(IGNORED_NUM_ARG))
+        .IgnoreArgument(1);
+
+    STRICT_EXPECTED_CALL(mocks, STRING_construct(TEST_SAS_CONNECTION_STRING));
+
+    STRICT_EXPECTED_CALL(mocks, connectionstringparser_parse(TEST_STRING_HANDLE))
+        .SetReturn(TEST_MAP_HANDLE);
+
+    STRICT_EXPECTED_CALL(mocks, Map_GetValueFromKey(TEST_MAP_HANDLE, (const char*)"SharedAccessKeyName"))
+        .SetReturn(TEST_CONST_CHAR_PTR);
+
+    STRICT_EXPECTED_CALL(mocks, Map_GetValueFromKey(TEST_MAP_HANDLE, (const char*)"DeviceId"))
+        .SetReturn(TEST_CONST_CHAR_PTR_NULL);
+
+    STRICT_EXPECTED_CALL(mocks, Map_GetValueFromKey(TEST_MAP_HANDLE, (const char*)"HostName"))
+        .SetReturn(TEST_CONST_CHAR_PTR);
+
+    STRICT_EXPECTED_CALL(mocks, Map_GetValueFromKey(TEST_MAP_HANDLE, (const char*)"SharedAccessKey"))
+        .SetReturn(TEST_CONST_CHAR_PTR_NULL);
+
+    STRICT_EXPECTED_CALL(mocks, Map_GetValueFromKey(TEST_MAP_HANDLE, (const char*)"SharedAccessSignature"))
+        .SetReturn(TEST_CONST_CHAR_PTR);
+
+    STRICT_EXPECTED_CALL(mocks, STRING_construct(TEST_CONST_CHAR_PTR));
+
+    STRICT_EXPECTED_CALL(mocks, STRING_TOKENIZER_create(TEST_STRING_HANDLE))
+        .SetReturn(TEST_STRING_TOKENIZER_HANDLE);
+
+    STRICT_EXPECTED_CALL(mocks, STRING_new())
+        .SetReturn(TEST_STRING_HANDLE);
+
+    STRICT_EXPECTED_CALL(mocks, STRING_new())
+        .SetReturn(TEST_STRING_HANDLE);
+
+    STRICT_EXPECTED_CALL(mocks, STRING_TOKENIZER_get_next_token(TEST_STRING_TOKENIZER_HANDLE, TEST_STRING_HANDLE, "."))
+        .SetReturn(0);
+
+    STRICT_EXPECTED_CALL(mocks, STRING_TOKENIZER_get_next_token(TEST_STRING_TOKENIZER_HANDLE, TEST_STRING_HANDLE, "0"))
+        .SetReturn(0);
+
+    EXPECTED_CALL(mocks, mallocAndStrcpy_s(IGNORED_PTR_ARG, IGNORED_PTR_ARG))
+        .SetReturn(0);
+
+    EXPECTED_CALL(mocks, mallocAndStrcpy_s(IGNORED_PTR_ARG, IGNORED_PTR_ARG))
+        .SetReturn(0);
+
+    EXPECTED_CALL(mocks, gballoc_malloc(IGNORED_NUM_ARG));
+
+    EXPECTED_CALL(mocks, STRING_c_str(TEST_STRING_HANDLE))
+        .SetReturn(TEST_CHAR_PTR);
+
+    EXPECTED_CALL(mocks, STRING_c_str(TEST_STRING_HANDLE))
+        .SetReturn(TEST_CHAR_PTR);
+
+    EXPECTED_CALL(mocks, mallocAndStrcpy_s(IGNORED_PTR_ARG, IGNORED_PTR_ARG))
+        .SetReturn(0);
+
+    EXPECTED_CALL(mocks, mallocAndStrcpy_s(IGNORED_PTR_ARG, IGNORED_PTR_ARG))
+        .SetReturn(0);
+
+    set_expected_calls_for_CreateFromConnectionString_cleanup(mocks);
+
+
+    // act
+    IOTHUB_SERVICE_CLIENT_AUTH_HANDLE result = IoTHubServiceClientAuth_CreateFromConnectionString(TEST_SAS_CONNECTION_STRING);
```

Comment:
We need a new API here named IoTHubServiceClientAuth_CreateFromSharedAccessSignature(). This is similar to our NodeJS Service API surface and I prefer it over overloading the current default-SharedAccessKey-behavior API.